docs/userguide/utils.md

```diff
@@ -0,0 +1,128 @@
+print(f"Expected data frame shape: {mem_estimate.comp_rep_shape}")
+```
+
+```{admonition} Estimate with Constraints
+:class: warning
+`estimate_product_space_size` currently does not include the influence of potential
+constraints in your search space as it is generally very hard to incorporate the effect
+of arbitrary constraints without actually buidling the entire space. Hence, you should
+always **treat the number you get as upper bound** of required memory. This can still be
+useful - for instance if your estimate already is several Exabytes, it is unlikely that
+most computers would be able to handle the result even if there are constraints present.
+```
+
+```{admonition} Influence of Continuous Parameters
+:class: info
+Continuous parameters fo not influence the size of the discrete search space part.
+Hence, they are ignored by the utility.
+```
+
+```{admonition} Memory During Optimization
+:class: warning
+`estimate_product_space_size` only estimates the memory required to handle the search
+space. **It does not estimate the memory required during optimization**, which can be
+of a similar magnitude, but generally depends on additional factors.
+```
+
+```{admonition} Effective Search Space Creation for Mixtures
+:class: tip
+If you run into issues creating large search spaces, as for instance for mixtures, you
+can try to use the [`SubspaceDiscrete.from_simplex`](baybe.searchspace.discrete.SubspaceDiscrete.from_simplex)
+constructor. Instead of creating the search space completely before filtering it down
+according to the constraints, this constructor includes the main mixture constraint
+already during the Cartesian product, requiring substantially less memory overall. In
+addition, BayBE can also be installed with an optional `polars` dependency (`pip install
+baybe[polars]`) that will utilize the more efficient machinery form polars for handling
+of the search space and its constraints.
+```
+
+## Reproducibility
+In some scenarios, for instance when testing your code setup, it can be useful to fix
+the random seeds for all relevant engines to generate reproducible results. BayBE offers
+the [`set_random_seed`](baybe.utils.random.set_random_seed) utility for this purpose:
+
+```python
+from baybe.utils.random import set_random_seed
+
+# Set the global random seed for all relevant engines
+set_random_seed(1337)
+
+# Assuming we have a prepared campaign
+campaign.recommend(5)
+```
+
+Setting the global random seed can be undesirable if there are other packages in your
+setup. For this, BayBE offers [`temporary_seed`](baybe.utils.random.temporary_seed):
+
+```python
+from baybe.utils.random import temporary_seed
+
+# Set the random seed for all relevant engines temporarily within the context
+with temporary_seed(1337):
+    campaign.recommend(5)
+```
+
+## Add Fake Target Measurements and Noise
+When creating test scripts, it is often useful to try the recommendation loop for a few
```

Comment:
I think this might now be the time to finally harmonize some of our terminology. Perhaps we should find a more consistent/precise name for `add_fake_results`, just like we have done in other places when we talk about `measurements` etc. So what is the most accurate wording here, since `results` is used nowhere else and is very generic? `add_fake_measurements`? `add_fake_targets_values`? Alternatively, we could also change it to something like `fantasize_measurements` or similar.

Reply:
renamed to `add_fake_measurements`
no deprecations for utilities